### DISCUSSION TITLE
iOS app cold start and deep links

### DISCUSSION THREAD

**Discussion by Nemesis19** [ionic-team/capacitor#5515] [Q&A]
Good evening,

I have a Webapp in Vuejs and I used Capacitor to make it native. I could set correctly all the deep link settings as provided in the docs but I have an issue in the situation of a "cold start" of the app. 

For "cold start" I mean that the app is not running and not present in the task manager.

I have implemented the `appUrlOpen` listener at the end of my main router.js file (where the vue-router gets inizialized) and it works correctly when the app has been already opened and visible in iOS task manager.

However in case of a cold start the listener doesn't work and the internal router just brings the user to the default route.

I tried to implement as per docs the `getLaunchUrl` method to retrieve the link that opened the app, but again it doesn't work. I did it inside the `async mounted` lifecycle in App.vue, hence the main component.

I wanted to see if it's a common problem of other frameworks, but Instagram opens a profile page even in cold start.

Is this a bug of Capacitor?

Thanks

**Comment by jonah-butler** [comment]
Has this been working for you in Chrome?
For me, Safari seems to recognize the app, regardless of its opened/closed state, and gives me the option to open it. The link was opening automatically several months ago, but since stopped. Safari prompts when hitting the deep link path, but Chrome does not work at all.

**Comment by Nemesis19** [comment]
I resolved the thing with this change:

```
App.addListener('appUrlOpen', function (event) {
	// Example url: https://beerswift.app/tabs/tabs2
	// slug = /tabs/tabs2
	const slug = event.url.split('.com').pop();

	// We only push to the route if there is a slug present
	if (slug) {
		window.requestAnimationFrame(() => {
			router.push({
				path: slug,
			});
		});
	}
});
```

so by using the `window.requestAnimationFrame(()`

**Reply by Nemesis19** [reply]
Honestly I never tried with other browsers different from Safari. I'll give it a try and let you know.

**Comment by jonah-butler** [comment]
Interesting. I'll investigate that when I know my appUrlOpen listener is getting hit. It has just stopped suddenly. No updates were made to the AndroidManifest.xml, any of the .well-known files or the apple-app-site-association file. So it's a little boggling. Yeah that would be cool if you let me know how it's working for you in Chrome. Appreciate that!

**Comment by fikretkurtiiev** [comment]
@Nemesis19 everything is good here?
I have non-worked Chrome, FF for iOS, Safari is good.

**Reply by Nemesis19** [reply]
I changed dev stack since my post, so I'm not sure I can help you 😅, sorry!
